protocol/communities/manager.go

```diff
@@ -81,6 +84,21 @@ type Manager struct {
 	historyArchiveDownloadTasks    map[string]*HistoryArchiveDownloadTask
 }
 
+type openseaClient interface {
+	FetchAllAssetsByOwnerAndContractAddress(owner gethcommon.Address, contractAddresses []gethcommon.Address, cursor string, limit int) (*opensea.AssetContainer, error)
+}
+
+type openseaClientBuilder interface {
+	NewOpenseaClient(uint64, string, *event.Feed) (openseaClient, error)
+}
+
+type defaultOpenseaBuilder struct {
+}
+
+func (b *defaultOpenseaBuilder) NewOpenseaClient(chainID uint64, apiKey string, feed *event.Feed) (openseaClient, error) {
```

Comment:
I don't think the opensea client is suppose to be exposed. There is a collectible package which I think is the one that we could leverage.

Reply:
Basically the code in `communities/manager.go` was already using `OpenseaClient` directly, it needed 

`FetchAllAssetsByOwnerAndContractAddress(owner gethcommon.Address, contractAddresses []gethcommon.Address, cursor string, limit int) (*opensea.AssetContainer, error)`

I made it an interface in order to swap out the implementation for testing (code wasn't tested and it's really hard to modify).

I am happy with anything that allows us to mock/stub in order to test (I generally don't like this kind of testing, but in this case the benefits are more than the drawbacks), where do you think it should go/which one we should be using?

Reply:
Ok, I will address this in a separate PR If it's ok, I need to add a few tests in any case, thank you!